Modified `meal`

kamilkrzyskow opened the pull request:
Fixed the `no_output` test to properly check if the output is empty.

Changed the input of `no_output` to `8:01` because the previous tests didn't check any value that went over a meal time with minutes only. Because of that, some students only used the `str` hours instead of following the specification to convert the `str` time to `float`.

Changed the order of execution for `no_output` and made other tests rely on the results of it, to prevent students from passing `meal` with a solution not per the specification.

kamilkrzyskow committed:
modified `meal`

bxngyn reviewed (commented):
Thanks for submitting a PR!

We typically follow a standard template when writing checks, where we use `check50.exists` as the decorator for all the functions and `check50.run`...`.exit()` to ensure valid execution. It seems you overrode all of the decorators with your new `test_no_output` function. The purpose of the function, which is to test for no output when given an invalid time, doesn't match the purpose of the other checks which is to test for valid output when given a valid time.

kamilkrzyskow commented:
It's been 3 years, so sorry if I come of as ignorant or forgetful.

`@check50.check(previous)` decorator blocks the execution of the decorated test if the `previous` test didn't pass. At the time of making PRs here my goal was to make the checker more "strict", so I moved the `no_output` test up the chain as it allowed to pass the assignment even without "perfect" code.

The idea was for the "strictness" of the assignment to lead the students to the proper "perfect" solution. I moved on from this ideology a wee bit over the years, as I now understand that there needs to be some leeway in the grading process to not destroy dreams 😸 

So I guess this will be closed similarly to:
- https://github.com/cs50/problems/pull/123

EDIT:
I see that there is a cycle to the PR pruning, which is close to ~summer time each year 😄 

Thanks for having a look and taking the time to review.

bxngyn commented:
No worries!

We appreciate your contribution to the course.
